src/en/elarcpage/src/eu/kanade/tachiyomi/extension/en/elarcpage/ElarcPage.kt

```diff
@@ -1,12 +1,81 @@
 package eu.kanade.tachiyomi.extension.en.elarcpage
 
 import eu.kanade.tachiyomi.multisrc.mangathemesia.MangaThemesia
+import eu.kanade.tachiyomi.network.GET
+import okhttp3.Interceptor
+import okhttp3.Response
+import org.jsoup.Jsoup
+import java.io.IOException
 
 class ElarcPage : MangaThemesia(
     "Elarc Toon",
     "https://elarctoon.com",
     "en",
-    "/readtoons98111",
 ) {
     override val id = 5482125641807211052
+
+    override val client = network.cloudflareClient.newBuilder()
+        .addInterceptor(::dynamicUrlInterceptor)
+        .build()
+
+    private var dynamicUrlUpdated: Long = 0
+    private val dynamicUrlValidity: Long = 10 * 60 // 10 minutes
+
+    private fun dynamicUrlInterceptor(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val timeNow = System.currentTimeMillis() / 1000
+
+        // Check if request requires an up-to-date URL
+        if (request.url.pathSegments[0] == mangaUrlDirectory.substring(1)) {
+            // Force update URL if required
+            if (timeNow - dynamicUrlUpdated > dynamicUrlValidity) {
+                client.newCall(GET(baseUrl)).execute()
```

Comment:
It gets the home page if mangaUrlDirectory is too old. This new request will also be intercepted, but it will skip this line because of line 29. When this new request loads it changes mangaUrlDirectory on line 65.